lib/factor/factorQuadratic.js

```diff
@@ -225,45 +227,88 @@ function factorSumProductRule(node, symbol, aValue, bValue, cValue, negate) {
     for (const pair of factorPairs) {
       if (pair[0] + pair[1] === bValue) {
         // To factor, we go through some transformations
-        // TODO: these should be actual substeps
-        // 1. Break apart the middle term into two terms using our factor pair (p and q):
-        //    e.g. ax^2 + bx + c -> ax^2 + px + qx + c
+        // 1. Break apart the middle term into two terms using our factor pair
+        //    (p and q): e.g. ax^2 + bx + c -> ax^2 + px + qx + c
         // 2. Consider the first two terms together and the second two terms
         //    together (this doesn't require any actual change to the expression)
         //    e.g. first group: [ax^2 + px] and second group: [qx + c]
         // 3. Factor both groups separately
         //    e.g first group: [ux(rx + s)] and second group [v(rx + s)]
         // 4. Finish factoring by combining the factored terms through grouping:
         //    e.g. (ux + v)(rx + s)
+        const substeps = [];
+        let status;
+
+        // STEP 1
         const pValue = pair[0];
         const qValue = pair[1];
-
-        // factor the first group to get u, r and s:
-        // u = gcd(a, p), r = a/u and s = p/u
+        const a = Node.Creator.constant(aValue);
+        const c = Node.Creator.constant(cValue);
+        const p = Node.Creator.constant(pValue);
+        const q = Node.Creator.constant(qValue);
+        const ax2 = Node.Creator.polynomialTerm(symbol, Node.Creator.constant(2), a);
+        const px = Node.Creator.polynomialTerm(symbol, null, p);
+        const qx = Node.Creator.polynomialTerm(symbol, null, q);
+
+        // SUBSTEP: ax^2 + bx + c -> ax^2 + px + qx + c
```

Comment:
mrah sorry for not commenting on this last time, but I think it's confusing to say substep after step - I think just merge them into one comment at the top

```
// SUBSTEP 1: ax^2 + bx + c -> ax^2 + px + qx + c
...
// SUBSTEP 2 ...
```

you see what I mean?